generative_ai/prompts/prompt_template.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+
+
+def prompt_template() -> str:
+    """Create prompt template"""
+
+    # [START generativeaionvertexai_prompt_template]
+    import vertexai
+    from vertexai.preview.prompts import Prompt
+
+    # Initialize vertexai
+    vertexai.init(project=PROJECT_ID, location="us-central1")
+
+    variables = [
+        {
+            "animal": ["""Eagels, Coyotes, Squirrels"""],
+            "activity": ["""eat berries, jump, fly"""],
+        },
+    ]
+    prompt = Prompt(
+        prompt_data=["Do {animal}{activity}"],  # Includes placeholders vor vars
+        model_name="gemini-1.5-flash-002",  # Model in use
+        variables=variables,  # Lists variables defined above
+        system_instruction=["You are a helpful zoolgist"]  # passes instructions for the llm
+        # optional - generation_config=generation_config,
+        # optional - safety_settings=safety_settings,
+    )
+    # Generates content using the assembled prompt.
+    responses = prompt.generate_content(
+        contents=prompt.assemble_contents(**prompt.variables[0]),
```

Comment:
I'm not sure what this is supposed to do here? Is it the recommended practice to unpack the variable like this?
Do you need to use `assemble_contents()` if the variables are already passed into the `Prompt` object?

Reply:
It's recommended to unpack the variable like this. With multiple variable sets, you will need a for loop and index into each set of variables, instead of just use the 0 index.